ext/vulnsrc/rocky/rocky.go

```diff
@@ -0,0 +1,201 @@
+	Description      string   `json:"description"`
+	AffectedProducts []string `json:"affectedProducts"`
+	Severity         string   `json:"severity"`
+	CVEs             []string `json:"cves"`
+	RPMs             []string `json:"rpms"`
+}
+
+type updater struct{}
+
+func init() {
+	vulnsrc.RegisterUpdater("rocky", &updater{})
+}
+
+func (u *updater) Update(datastore vulnsrc.DataStore) (resp vulnsrc.UpdateResponse, err error) {
+	log.WithField("package", "Rocky").Info("Start fetching vulnerabilites")
+
+	// Download JSON
+	r, err := httputil.GetWithUserAgent(url)
+	if err != nil {
+		log.WithError(err).Error("could now download Rocky's update")
+		return resp, commonerr.ErrCouldNotDownload
+	}
+	defer r.Body.Close()
+
+	if !httputil.Status2xx(r) {
+		log.WithField("StatusCode", r.StatusCode).Error("Failed to update Rocky")
+		return resp, commonerr.ErrCouldNotDownload
+	}
+
+	// Get the SHA-1 of the latest JSON data
+	latestHash, err := datastore.GetKeyValue(updateFlag)
+	if err != nil {
+		return resp, err
+	}
+
+	// Parse JSON
+	resp, err = buildResponse(r.Body, latestHash)
+	if err != nil {
+		return resp, err
+	}
+
+	return resp, nil
+}
+
+func (u *updater) Clean() {}
+
+func buildResponse(jsonReader io.Reader, latestKnownHash string) (resp vulnsrc.UpdateResponse, err error) {
+	hash := latestKnownHash
+
+	jsonSHA := sha1.New()
+	teedJSONReader := io.TeeReader(jsonReader, jsonSHA)
+
+	// Unmarshal JSON
+	var data jsonData
+	err = json.NewDecoder(teedJSONReader).Decode(&data)
+	if err != nil {
+		log.WithError(err).Error("could not unmarshal Rocky's JSON")
+		return resp, commonerr.ErrCouldNotParse
+	}
+
+	// Calculate the hash and skip updating if the hash has been seen before.
+	hash = hex.EncodeToString(jsonSHA.Sum(nil))
+	if latestKnownHash == hash {
+		log.WithField("package", "Rocky").Debug("no update")
+		return resp, nil
+	}
+
+	// Extract vulnerability data from Rocky's JSON schema.
+	resp.Vulnerabilities = parseRockyJSON(&data)
+
+	return resp, nil
+}
+
+func parseRockyJSON(data *jsonData) (vulnerabilities []database.Vulnerability) {
+	mvulnerabilities := make(map[string]*database.Vulnerability)
+
+	for _, advisory := range data.Advisories {
+		if strings.ToLower(advisory.Type) != "security" {
+			continue
+		}
+
+		vulnName := advisory.Name
+		vulnerability, vulnerabilityAlreadyExists := mvulnerabilities[vulnName]
+
+		if !vulnerabilityAlreadyExists {
+			var subCVEs []string
+			for _, cve := range advisory.CVEs {
+				cveID := cve[strings.LastIndex(cve, ":::")+3:]
+				if strings.HasPrefix(cveID, "CVE-") {
+					subCVEs = append(subCVEs, cveID)
+				}
+			}
+
+			vulnerability = &database.Vulnerability{
+				Name:        vulnName,
+				Link:        fmt.Sprintf(linkFormat, vulnName),
+				Severity:    normalizeSeverity(advisory.Severity),
+				Description: advisory.Description,
+				SubCVEs:     subCVEs,
+			}
+		}
+
+		for _, rpmName := range advisory.RPMs {
+			// FixMe: This is not ideal and likely error prone.
```

Comment:
I think the `affectedProducts` may help? Though, at the moment, all I see is 265 instances of `Rocky Linux 8`. Not sure if they'll all just be `Rocky Linux X` or something else is possible. Are there any docs which discuss the format? Also, you mentioned a v2 soon. Would that help here?

Reply:
Yeah, I'm not entirely sure about that yet either. I did find this, https://docs.rockylinux.org/books/admin_guide/13-softwares/#rpm-redhat-package-manager. But I'll dig into this further.

The v2 api will hopefully have this data broken down already with specifics on affected distro versions. Will make some inquiries on the new api.